slatedb-cli/src/main.rs

```diff
@@ -22,6 +25,19 @@ async fn main() -> Result<(), Box<dyn Error>> {
     let args: CliArgs = parse_args();
     let path = Path::from(args.path.as_str());
     let object_store = admin::load_object_store_from_env(args.env_file)?;
+
+    let cancellation_token = CancellationToken::new();
```

Comment:
i think this cancellation token is worth promoting in all critical structures :+1:. `context` is something I miss a lot from Go when moving to Rust, it forces us to think clearly about how the canceling is handled in the early design and provides a standardized approach for it.

Reply:
They are not the same as Go's context, I would not prematurely optimize the code to pass them everywhere like you'd do in Go. As far as I can tell, this is only needed to run the GC in the background at the moment, if there are more use cases in the future, we can pass it along. I put it at the top of this file, so it's easier to spot and reuse when necessary.

Reply:
make sense :+1: